vertx-web/src/main/java/io/vertx/ext/web/handler/SessionHandler.java

```diff
@@ -269,4 +270,13 @@ static SessionHandler create(SessionStore sessionStore) {
    * @return the session
    */
   Session newSession(RoutingContext context);
+
+  /**
+   * Set the user for the session
+   *
+   * @param context the routing context
+   * @param user the user
+   * @return the user
+   */
+  User setUser(RoutingContext context, User user);
```

Comment:
Looking at the implementation of this method, I think the signature should be changed:

```java
Future<Void> setUser(RoutingContext context, User user);

// and
@Fluent
default Session setUser(RoutingContext context, User user, Handler<AsyncResult<Void>> handler) {
  setUser(contex, user).onComplete(handler);
  return this;
}
```

Because using it this way would allow us not to discard the result of the asynchronous operation that is happening inside the method implementation.

Reply:
I'm confused here, `return this` from the `SessionHandler` would not give you a `Session` but rather a `SessionHandler` instance right?